chia/full_node/mempool_manager.py

```diff
@@ -351,9 +377,15 @@ async def add_spend_bundle(
             self.mempool.add_to_pool(item)
             self.mempool.remove_from_pool(remove_items, MempoolRemoveReason.CONFLICT)
             return item.cost, MempoolInclusionStatus.SUCCESS, None
+        elif err is Err.MEMPOOL_CONFLICT and item is not None:
+            # The transaction has a conflict with another item in the
+            # mempool, put it aside and re-try it later
+            self._conflict_cache.add(item)
+            return item.cost, MempoolInclusionStatus.PENDING, err
         elif item is not None:
-            # There is an error,  but we still returned a mempool item, this means we should add to the pending pool.
-            self.potential_cache.add(item)
+            # The transasction has a height assertion and is not yet valid.
+            # remember it to try it again later
+            self._pending_cache.add(item)
```

Comment:
Can an item enter the `_pending_cache` for a reason other than a height assertion?

Reply:
it's not supposed to be possible, and I don't see how it would be possible. we only add it to one or the other. and the only way an item is coming back here is if it was first drained from one of the caches.